src/SFML/Graphics/Shader.cpp

```diff
@@ -646,9 +646,12 @@ void Shader::setUniformArray(const std::string& name, const Glsl::Mat3* matrixAr
 {
     const std::size_t matrixSize = 3 * 3;
 
-    std::vector<float> contiguous(matrixSize * length);
+    std::vector<float> contiguous;
+    contiguous.reserve(matrixSize * length);
```

Comment:
Why use `.reserve` over the `vector` constructor which does the same thing?

Reply:
You probably confuse it with `resize()`. The `vector` constructor accepting a number default-constructs those elements, doesn't just allocate space.

But what was the idea behind this change? Simpler code?

Reply:
I first though `copyMatrix(const float* source, std::size_t elements, float* dest)` was not needed anymore when editing Glsl.inl/cpp and removed it. Then this code was broken. I replaced it with a loop without thinking too much about it.
Using `reserve` instead of `resize` or the constructor with a size allows me to use `push_back` to take care of the destination index instead of computing it myself in the loop below. The idea was me being lazy.
Investigating some more with [a small benchmark](https://www.quick-bench.com/q/K7iEBSn4GxLy_uPqDJKbthwUY2o), it seems that using `push_back` is much slower than `std::copy` or computing the destination index, so this code will change.

Reply:
I went with `copyMatrix` as before.